### DISCUSSION TITLE
Brew Search Broken Not working (Calling depends_on :x11 is disabled! Use depends_on specific X11 formula(e) instead.)

### DISCUSSION THREAD

**Discussion by XxUnkn0wnxX** [Homebrew/discussions#842] [Everyday usage]
when I do brew search python or brew search man I keep getting the following output 

```
Error: Calling depends_on :x11 is disabled! Use depends_on specific X11 formula(e) instead.
Please report this issue to the homebrew/cask tap (not Homebrew/brew or Homebrew/core), or even better, submit a PR to fix it:
  /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/gnubg.rb:17
```

is this a bug I need to report?
this is frustrating now I cannot even use brew search at all...

**Comment by SMillerDev** [comment]
I think `brew config` will tell you your cask repo is pretty outdated.
You can run `brew update` to fix that.

**Reply by XxUnkn0wnxX** [reply]
```
HOMEBREW_VERSION: 3.0.1-82-g431e448
ORIGIN: https://github.com/Homebrew/brew
HEAD: 431e448c9e85ec6f56ef518e4e443509e214b10e
Last commit: 48 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 80e1e34bb34368525f1695a22a72f1d122a17c8e
Core tap last commit: 38 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.K7OAd8TvW3/org.macosforge.xquartz:0
HOMEBREW_EDITOR: /usr/local/bin/nano
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 12.0 build 1200
Git: 2.30.1 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.1-x86_64
CLT: N/A
Xcode: 12.4
XQuartz: 2.7.11 => /opt/X11
```

i dont think it is, I just updated brew & I still get the issue

**Reply by SMillerDev** [reply]
Oh, that file isn't actually part of the homebrew-cask repo. In that case, can you post the output of `brew doctor`?

**Reply by XxUnkn0wnxX** [reply]
```
Warning: Unknown support status

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed with `brew diy`.
You should find replacements for the following formulae:
  unrar
  openssl@1.0

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  libav
  lua@5.1
  polipo
  spdylay

Warning: You have installed Bitdefender. The "Traffic Scan" option interferes with
Homebrew's ability to download packages. See:
  https://github.com/Homebrew/brew/issues/5558

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /usr/local/opt/pkg-config

Warning: libiconv files detected at a system prefix other than /usr.
Homebrew doesn't provide a libiconv formula, and expects to link against
the system version in /usr. libiconv in other prefixes can cause
compile or link failure, especially if compiled with improper
architectures. macOS itself never installs anything to /usr/local so
it was either installed by a user or some other third party software.

tl;dr: delete these files:
  /usr/local/include/iconv.h
  /usr/local/include/iconv.h

Warning: Putting non-prefixed coreutils in your path can cause GMP builds to fail.

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  lua@5.1
  openssl@1.0
  python@3.9
```
  
openssl I'm using via DIY & python@3.9 I want to keep a keg only as it overwrites my links for python@3.8 as I need it globally for the software I use.. & I cannot remove python@3.9 as it's depended on other formulae.

with brew update or even the reset command it doesn't do a good job at deleting old formula...
is there a command to get brew to completely reset all local formula? like, delete all the .rb fills on my system & get brew to re-clone all of them?

also cant I just delete `gnubg.rb` ?

**Reply by SMillerDev** [reply]
> with brew update or even the reset command it doesn't do a good job at deleting old formula...
is there a command to get brew to completely reset all local formula? like, delete all the .rb fills on my system & get brew to re-clone all of them?

You can `brew untap homebrew/cask` and `brew tap homebrew/cask`

> also cant I just delete gnubg.rb ?

Possibly, but it might not be the only affected file.

**Comment by gromgit** [comment]
The `gnubg` cask was [removed over a year ago](https://github.com/Homebrew/homebrew-cask/pull/72428), so it seems your cask tap is _way_ out of date.

Have you been getting errors on `homebrew/homebrew-cask` during your `brew updates`? If so, and a `brew update-reset` doesn't fix it, you'll probably have to do the following:
```
brew untap homebrew/cask
brew tap homebrew/cask
```

**Reply by XxUnkn0wnxX** [reply]
umm it doesnt cozz: 

```
Error: Refusing to untap homebrew/cask because it contains the following installed formulae or casks:
cursorcerer
db-browser-for-sqlite
fugu
mono-mdk
mpv
provisionql
qlcommonmark
qlimagesize
qlmarkdown
qlstephen
qlvideo
quicklook-csv
quicklook-json
quicklook-pat
quicklookapk
quicklookase
suspicious-package
xquartz
yacreader
```

**Reply by gromgit** [reply]
Did you run `brew update-reset` before trying to untap `homebrew-cask`? If not, try that first, then `brew search python` again.

**Reply by XxUnkn0wnxX** [reply]
yes i have done the reset, still nothing
I just deleted the affect rb file & it searching again.. however, I cannot do untap due to the installed formulae

**Reply by gromgit** [reply]
Deleting `gnubg.rb` only treats a symptom, not the underlying problem. I wouldn't be surprised if some (or all) of your installed casks are out of date, because the cask tap hadn't been successfully updated in a long time.

Fortunately, you can force the untap:
```
brew untap --force homebrew/cask
brew tap homebrew/cask
```
